lib/_http_client.js

```diff
@@ -636,7 +636,11 @@ ClientRequest.prototype.onSocket = function onSocket(socket) {
 function onSocketNT(req, socket) {
   if (req.aborted) {
     // If we were aborted while waiting for a socket, skip the whole thing.
-    socket.emit('free');
+    if (req.socketPath || !req.agent) {
+      socket.destroy();
+    } else {
+      socket.emit('free');
+    }
```

Comment:
@bnoordhuis I guess you were referring to https://github.com/nodejs/node/blob/77be180f49096c2789c08d31744f5ef16fe340a1/lib/_http_client.js#L296-L299. I'm not sure either.